middleware/recoverer.go

```diff
@@ -21,7 +21,12 @@ import (
 func Recoverer(next http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
 		defer func() {
-			if rvr := recover(); rvr != nil && rvr != http.ErrAbortHandler {
+			if rvr := recover(); rvr != nil {
+				if rvr == http.ErrAbortHandler {
```

Comment:
what difference does this provide to move the `rvr` condition testing `http.ErrAbortHandler` below..? I'm wondering how this solves your issue? thanks for explaining

Reply:
It solves the issue because before this change the panic is recovered and so the http connection is not aborted (the recoverer sends a 500 status code). 

After this change we have:

```
if rvr == http.ErrAbortHandler {
	// we don't recover http.ErrAbortHandler so the response
	// to the client is aborted
	panic(rvr)
}
```

and so if the error is `http.ErrAbortHandler` the panic is not recovered anymore.